blackbox/test_jmx.py

```diff
@@ -156,6 +156,31 @@ def test_mbean_node_id(self):
         self.assertEqual(stderr, '')
         self.assertNotEqual(stdout.rstrip(), '', 'node id must not be empty')
 
+    def test_mbean_shards(self):
+        jmx_client = JmxClient(JMX_PORT)
+        with connect(enterprise_crate.http_url) as conn:
+            c = conn.cursor()
+            c.execute('''create table test(id integer) clustered into 1 shards with (number_of_replicas=0)''')
+            stdout, stderr = jmx_client.query_jmx(
+                'io.crate.monitoring:type=NodeInfo',
+                'ShardStats'
+            )
+            result = list((line.strip() for line in stdout.split('\n') if line.strip()))
```

Comment:
```suggestion
            result = [line.strip() for line in stdout.split('\n') if line.strip()]
```

It is more common to use a list comprehension directly instead of calling `list` on a generator expression.